shell/platform/linux/key_mapping.cc

```diff
@@ -0,0 +1,467 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "key_mapping.h"
+
+#include <glib.h>
+#include <map>
+
+#include "flutter/shell/platform/linux/fl_key_embedder_responder_private.h"
+
+// DO NOT EDIT -- DO NOT EDIT -- DO NOT EDIT
+// This file is generated by
+// flutter/flutter@dev/tools/gen_keycodes/bin/gen_keycodes.dart and should not
+// be edited directly.
+//
+// Edit the template dev/tools/gen_keycodes/data/gtk_key_mapping_cc.tmpl
+// instead. See dev/tools/gen_keycodes/README.md for more information.
+
+// Insert a new entry into a hashtable from uint64 to uint64.
+//
+// Returns whether the newly added value was already in the hash table or not.
+static bool insert_record(GHashTable* table, guint64 xkb, guint64 fl_key) {
+  return g_hash_table_insert(table, uint64_to_gpointer(xkb),
+                             uint64_to_gpointer(fl_key));
+}
+
+void initialize_xkb_to_physical_key(GHashTable* table) {
```

Comment:
I know this is using gtk, but why not use a std::map here for all of these tables?  You can initialize them statically that way, and won't need to call an initializer function.

Reply:
I'm inclining to use full C (instead of C++) in the Linux embedding. I doubt if it's necessary, but I think that's we're committing to since we've been using GTK class (instead of C++ class) all over.

Reply:
I don't think you need to do that: there's already C++ in the linux embedding, because it uses the embedder API. I agree that in general we should try to use GTK classes for the GTK embedding, but in this case, it means we can't statically define these data structures, and there's no need to waste startup time setting them up just to use a C/GTK interface.

Reply:
I'm actually running into something similar in the iOS implementation: because the KeyCodes.mm file uses NSDictionary, those structures are not always initialized when loaded in XCode as a dylib, and it crashes when in the debugger. So I'm converting them to use std::map, which has no such issue.

Reply:
The embedder API is pure C AFAIK, otherwise we wouldn't need to use pure function and user_data instead of the convenient lambdas.

Reply:
Yes, the API is C-compatible, but the implementation is C++.

Reply:
Still, why do we keep using GTK class everywhere then? Almost all classes can be converted to C++ classes since they won't be passed to GTK APIs anyway, and we'll have so much cleaner code.

Reply:
How would using a C++ implementation here restrict users of the Linux embedding? Isn't this just an implementation detail of the key handling?

Reply:
> Still, why do we keep using GTK class everywhere then? 

As far as I know, it is only a style thing: when in GTK land, do as GTK does, that kind of thing. I think the idea is that someone maintaining the GTK code might be more familiar with GTK style than some hybrid (and it would be a hybrid, since we'd have to transform to GTK structs somewhere). Style shouldn't slow things down unnecessarily, though, IMO.

Reply:
I'll change the map to std::map, but seriously, the tediousness of GTK classes is beyond acceptance to the point that I'm missing Objective C classes. We might want to reconsider this decision some day.

Reply:
Yes, I agree. I almost think that we could abstract a lot of this code into some common C++ classes that take delegates for the platform specific work.

Reply:
I say "almost" because I'm pretty sure there are some snags in that plan. But there is a lot of similar logic in the implementations now.

Reply:
Keyboard manager is the most likely class to be extracted, but then we have iOS. Embedder responder is much less likely: the maps are not shared, the synthesization logic is very different, and the rest is mostly trivial.